BlueDBOnDisk/src/main/java/org/bluedb/disk/backup/BackupManager.java

```diff
@@ -76,10 +78,21 @@ protected void backupNewChangesToTempDirectory(List<ReadWriteCollectionOnDisk<?>
 		}
 	}
 
-	private void copyMetaData(ReadWriteCollectionOnDisk<?> collection, Path tempFolder) throws BlueDbException {
-		Path srcPath = collection.getMetaData().getPath();
-		Path dstPath = translatePath(dbPath, tempFolder, srcPath);
-		FileUtils.copyDirectoryWithoutLock(srcPath, dstPath);
+	private void copyMetadata(ReadWriteCollectionOnDisk<?> collection, Path tempFolder) throws BlueDbException {
+		try (Stream<Path> paths = Files.walk(collection.getMetaData().getPath())) {
+			paths
+					.filter(Files::isRegularFile)
+					.forEach(srcPath -> {
+						Path destPath = translatePath(dbPath, tempFolder, srcPath);
+						try {
+							collection.getFileManager().makeUnencryptedCopy(srcPath, destPath);
+						} catch (BlueDbException e) {
+							throw new UncheckedBlueDbException(e);
+						}
+					});
+		} catch (UncheckedBlueDbException | IOException e) {
+			throw new BlueDbException("Failed to backup metadata", e);
+		}

```

Comment:
I'll verify this in testing soon. I am pretty sure I can throw an unchecked exception within a lambda and that the problem with lambdas has to do with checked exceptions, but I may be wrong. I'll update if needed though, hopefully it works though!

Reply:
Tested and verified this to be working as expected. Did a bit of Googling to understand the issue more thoroughly in case you are curious:

Essentially, the lambda functions are driven by interfaces such as Consumer and Supplier that don't throw any checked exceptions. Throwing one within the inline implementation of that interface breaks the contract of that interface and fails to compile.
There are some fancier solutions like creating wrapper interfaces that can throw these exceptions but I don't think they really justify the extra code for a case like this when this code feels pretty readable as well.